smart-contracts/contracts/PublicLock.sol

```diff
@@ -221,20 +221,42 @@ contract PublicLock is ILockCore, ERC165, IERC721, IERC721Receiver, Ownable {
   }
 
   /**
-   * @dev Called by owner to wiwthdraw all funds from the lock.
+   * @dev Called by owner to withdraw all funds from the lock.
    * TODO: consider allowing anybody to trigger this as long as it goes to owner anyway?
-   * TODO: consider partial withdraws?
    * TODO: check for re-entrency?
    */
-  function withdraw(
-  )
+  function withdraw()
     external
     onlyOwner
   {
     uint256 balance = address(this).balance;
     require(balance > 0, "Not enough funds");
+    _withdraw(balance);
+  }
+
+  /**
+   * @dev Called by owner to partially withdraw funds from the lock.
+   * TODO: consider allowing anybody to trigger this as long as it goes to owner anyway?
+   * TODO: check for re-entrency?
+   */
+  function partialWithdraw(uint _amount)
```

Comment:
Should we set a lower bound on withdrawal `_amount`? it could be less than tx fee...

Reply:
And/or a limit on the minimum lock balance which allows partial withdrawals?

Reply:
I guess we should not do that at the contract level but we can do that in the UI?

Reply:
yes, I think that's fine here.